src/status_im2/contexts/wallet/send/select_address/view.cljs

```diff
@@ -41,70 +44,155 @@
-          :address-regex         constants/regx-address
-          :scanned-value         scanned-address
-          :on-detect-ens         on-detect-address-or-ens
-          :on-detect-address     on-detect-address-or-ens
-          :on-change-text        (fn [text]
+  [input-value input-focused?]
+  (fn []
+    (let [scanned-address       (rf/sub [:wallet/scanned-address])
+          send-address          (rf/sub [:wallet/send-address])
+          valid-ens-or-address? (boolean (rf/sub [:wallet/valid-ens-or-address?]))]
+      [quo/address-input
+       {:on-focus              (fn []
+                                 (when (empty? @input-value)
+                                   (rf/dispatch [:wallet/clean-local-suggestions]))
+                                 (reset! input-focused? true))
+        :on-blur               #(reset! input-focused? false)
+        :on-scan               (fn []
+                                 (rn/dismiss-keyboard!)
+                                 (rf/dispatch [:open-modal :scan-address]))
+        :ens-regex             constants/regx-ens
+        :address-regex         constants/regx-address
+        :scanned-value         (or send-address scanned-address)
+        :on-detect-ens         #(debounce/debounce-and-dispatch
+                                 [:wallet/validate-ens %]
+                                 300)
+        :on-detect-address     #(debounce/debounce-and-dispatch
+                                 [:wallet/validate-address %]
+                                 300)
+        :on-change-text        (fn [text]
+                                 (let [starts-like-eth-address (re-matches
+                                                                constants/regx-address-fragment
+                                                                text)]
                                    (when-not (= scanned-address text)
                                      (rf/dispatch [:wallet/clean-scanned-address]))
-                                   (reset! input-value text))
-          :on-clear              #(rf/dispatch [:wallet/clean-scanned-address])
-          :valid-ens-or-address? @valid-ens-or-address?}]))))
+                                   (if starts-like-eth-address
+                                     (rf/dispatch [:wallet/fetch-address-suggestions text])
+                                     (rf/dispatch [:wallet/clean-local-suggestions]))
+                                   (reset! input-value text)))
+        :valid-ens-or-address? valid-ens-or-address?}])))
+
+(defn- ens-linked-address
+  [{:keys [address networks theme]}]
+  [quo/text
+   {:size  :paragraph-2
+    :style {:padding-horizontal 12
+            :padding-top        4}}
+   (map (fn [network]
+          ^{:key (str network)}
+          [quo/text
+           {:size  :paragraph-2
+            :style {:color (colors/resolve-color network theme)}}
+           (str (subs (name network) 0 3) ":")])
```

Comment:
let's use the structure I commented above - if you use the sub `:wallet/network-details` you will see the shape for networks and we don't need to do this manipulation here 👍